.github/workflows/build-main.yml

```diff
@@ -55,48 +59,43 @@ jobs:
       with:
         run: python -m pytest --cov-report xml --cov=napari_aicsimageio napari_aicsimageio/tests/
     - name: Upload codecov
-      uses: codecov/codecov-action@v1
+      uses: codecov/codecov-action@v3
 
   lint:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v3
     - name: Set up Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v4
       with:
         python-version: 3.9
     - name: Install Dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install .[test]
-    - name: Lint with flake8
-      run: |
-        flake8 napari_aicsimageio --count --verbose --show-source --statistics
-        isort napari_aicsimageio --check-only
-        mypy napari_aicsimageio
-        black --check --exclude vendor napari_aicsimageio
+        pip install pre-commit
+    - name: Lint with pre-commit
+      run: pre-commit run --all-files
```

Comment:
using pre-commit is a great way to single-source what it means to check that everything you care about is all set.  It can be executed directly (even without running `pre-commit install` first).

Even better than this, would be to use https://pre-commit.ci, which will run faster (and removes the need to explicitly state it in your github yaml)

Reply:
This is one of those things I am still hesitant about. Don't know why. I think it's because I work with a lot of junior devs and I want them to _learn_ linting, formating, type checks and stuff and _why_ something changed. Granted, no junior devs work on this project I guess but would still prefer to leave it how you have it now.

Unless you have an argument against me on that one haha.

Reply:
sure, not using `pre-commit.ci` is totally reasonable.  I like it for me, since I care less about teaching every PR contributor how to fix stuff just how _I_ like it in my repo... but i do get that inclination too.  (but even for stuff like black?)

but yeah, `pre-commit.ci` is orthogonal here.  `pre-commit` itself is still a great way to single source all of your linting/formatting workflow, even if someone never runs `pre-commit install` locally.
 
one other nice thing that `pre-commit.ci` will do for you is run `pre-commit autoupdate` and open PRs periodically to update your linting/formatting deps.